### DISCUSSION TITLE
Could not notarize package with message ''

### DISCUSSION THREAD

**Discussion by dean-ivre** [fastlane/fastlane#20899] [General]
I am getting this issue for using the fastlane notarize for macOS application.
I have set my apple id credentials and app specific password , the error coming through is _**Failed to get authorization for username '--password' and password.**_, but I have dont know the reason for this not working
I have tried using atool and notarytool

**Comment by seansaleh** [comment]
Try running the notarize action with the flags `verbose: true`, `print_log: true`,
That should give you some better logs

If those logs don't help you can get even further logs with the command `xcrun notarytool log`
To build up the arguments you need to find four things
1. You need an api key .p8 file, and get the path to it
2. The key-id (in the example below it is `C23keyidexample`)
3. The issuer (in the example below it is `2efeb8--guid-that-is-the-issuer-id`)
4. The notarizazion run's id (in the example below it is `e20c0438-guid-that-is-the-log-id`)

You can find these last three from the notarize action's logs


```
INFO [2022-12-13 19:08:29.81]: ----------------------
INFO [2022-12-13 19:08:29.81]: --- Step: notarize ---
INFO [2022-12-13 19:08:29.81]: ----------------------
INFO [2022-12-13 19:08:29.81]: $ ditto -c -k --rsrc --keepParent "YourApp.app" "YourApp.app.zip"
INFO [2022-12-13 19:09:34.25]: $ xcrun notarytool submit YourApp.app.zip --output-format json --wait --key /var/folders/h7/1x743543534xxxxx/T/20221213-1010-b1234x --key-id C23keyidexample --issuer 2efeb8--guid-that-is-the-issuer-id
INFO [2022-12-13 19:14:09.32]: ▸ {"status":"Invalid","message":"Processing complete","id":"e20c0438-guid-that-is-the-log-id"}
WARN [2022-12-13 19:14:09.33]: Lane Context:
INFO [2022-12-13 19:14:09.33]: {:PLATFORM_NAME=>nil, :LANE_NAME=>"notar"}
ERROR [2022-12-13 19:14:09.33]: Could not notarize package with message ''
```

Then you can take those and run a command like so `xcrun notarytool log e20c0438-guid-that-is-the-log-idx --key <path to your api kiey .p8 file> --key-id C23keyidexample --issuer 2efeb8--guid-that-is-the-issuer-id`
_Note that the issuer and log ids change each time you run the notarize action_


For us, we found that we hadn't signed the binaries properly
